src/backend/distributed/utils/maintenanced.c

```diff
@@ -258,6 +264,28 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 		 * tasks should do their own time math about whether to re-run checks.
 		 */
 
+		/* the config value of 1000 means disabling the deadlock detection */
+		if (DistributedDeadlockDetectionTimeoutFactor != 1000)
+		{
+			StartTransactionCommand();
+			foundDeadlock = CheckForDistributedDeadlocks();
+			CommitTransactionCommand();
+		}
+
+		/*
+		 * If we find any deadlocks, run the distributed deadlock detection
+		 * more often since it is likely that that are other deadlocks needs to
```

Comment:
Just to understand better, why do we think in that way? I'll bring this up on our meeting to understand more.